dockertest/images.py

```diff
@@ -430,6 +430,17 @@ def default_image(self):  # pylint: disable=R0201
         # DICLS may have overriden this method
         return DockerImage.full_name_from_defaults(defaults)
 
+    def default_image_exist_check(self):
+        """
+        Retrieve default image if it does not exist in
+         ``self.filter_list_full_name()``
+        :return: ``autotest.client.utils.CmdResult`` instance
+        """
+        img_list = self.list_imgs()
+        default_image = self.full_name_from_defaults()
+        if not self.filter_list_full_name(img_list, default_image):
+            return self.docker_cmd("pull %s" % default_image, self.timeout)
```

Comment:
I understand why you want to return a CmdResult.  However, this API is intended for convenience, and not to be used for test subjects (so 'docker pull' in this case).  Therefore, I think it may be more practical to return the default image's ``DockerImage`` instance instead (the 0-th item from filter-list...() call).  This way the caller has ready-access to the image ID, size, broken-down name, etc.  What is your opinion?  Is there a reason returning CmdResult is better idea?